Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -751,14 +746,20 @@ SMILTime SVGSMILElement::simpleDuration() const
     return std::min(dur(), SMILTime::indefinite());
 }
 
+static void insertSorted(Vector<SMILTimeWithOrigin>& list, SMILTimeWithOrigin time)
+{
+    auto position = std::lower_bound(list.begin(), list.end(), time);
+    list.insert(position - list.begin(), time);
```

Comment:
Probably valuable to assert the list is already sorted in debug builds, as long as no one turns it on for release builds and ruins the optimization. I also don’t think the local variable is particularly helpful. I would do something like one of these:
```
ASSERT(std::is_sorted(list.begin(), list.end());
list.insert(std::lower_bound(list.begin(), list.end(), time) - list.begin(), time);
```
or
```
ASSERT(std::is_sorted(list.begin(), list.end());
auto position = std::lower_bound(list.begin(), list.end(), time) - list.begin();
list.insert(position, time);
```

Reply:
I opted for first option. Bit more cleaner. :-)